internal/clients/http/azure/azure_client.go

```diff
@@ -10,61 +10,55 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
 	"github.com/RHEnVision/provisioning-backend/internal/clients"
 	"github.com/RHEnVision/provisioning-backend/internal/config"
-	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
-	"github.com/rs/zerolog"
 )
 
-type azureClient struct {
+type client struct {
 	subscriptionID string
 	credential     *azidentity.ClientSecretCredential
 }
 
 func init() {
-	clients.GetAzureClient = newAzureClient
+	clients.GetCustomerAzureClient = newCustomerClient
 }
 
-func logger(ctx context.Context) zerolog.Logger {
-	return ctxval.Logger(ctx).With().Str("client", "azure").Logger()
-}
-
-func newAzureClient(ctx context.Context, auth *clients.Authentication) (clients.Azure, error) {
+func newCustomerClient(ctx context.Context, auth *clients.Authentication) (clients.Azure, error) {
 	opts := azidentity.ClientSecretCredentialOptions{}
 	identityClient, err := azidentity.NewClientSecretCredential(config.Azure.TenantID, config.Azure.ClientID, config.Azure.ClientSecret, &opts)
 	if err != nil {
 		return nil, fmt.Errorf("unable to init Azure credentials: %w", err)
 	}
 
-	return &azureClient{
```

Comment:
I would keep azureClient for consistency or change in ec2 and GCP to client as well

Reply:
Because each implementation lives its in own package, you can just use `client` and there will be no conflicts. Whatever you like.